src/modules/date/date-string.transformations.ts

```diff
@@ -29,4 +29,24 @@ export class DateStringTransformations {
     }
     return dateOnlyString + 'T00:00:00Z';
   }
+
+  getEffectiveDate(dateOnlyString: DateString): DateString {
+    const today = new Date();
+    if (new Date(dateOnlyString) < today) {
+      return this.addTimeToDateOnlyString(dateOnlyString);
+    } else {
+      const todayAsDateOnlyString = this.removeTimeIfExists(today.toISOString());
+      return this.addTimeToDateOnlyString(todayAsDateOnlyString);
+    }
+  }
+
+  getExpirationDate(dateOnlyString: DateString | null): DateString | null {
+    if (dateOnlyString) {
+      return this.addTimeToDateOnlyString(dateOnlyString);
+    } else if (PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate) {
+      return PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate;
+    } else {
+      return null;
+    }
```

Comment:
I don't think this should be in `dateTransformations` because it is very specific to deal investors